### PR TITLE
WT-9863 - Downgrade assertion in __hs_delete_reinsert_from_pos

### PR DESCRIPTION
**Pull request by ajmorton**
This assertion was triggering in mongo tests and causing a crash and core dump. However, the impact of this assertion not firing would be to add an invisible update to the history store which would be quickly cleaned up, has no impact on data integrity, and is preferable to a crash in release mode.

As such, downgrade this assertion to only fire in diagnostic mode. Investigation into the root cause will continue in WT-9846

### CODE REVIEW

**Review by quchenhao — APPROVED** [review]
lgtm

File: src/history/hs_rec.c [hunk 1]
```diff
@@ -900,6 +900,9 @@ __hs_delete_reinsert_from_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, ui
     WT_UNUSED(key);
 #endif
 
+    /* This function should only accept mixed mode timestamped updates. */
+    WT_ASSERT(session, ts == 1 || ts == WT_TS_NONE);
```
Comment (agorrod) [inline src/history/hs_rec.c:904]: Could you add a TODO here it turn this back into a WT_ASSERT_ALWAYS? I don't want to forget to switch it back - we should be able to enforce that.

Comment (ajmorton) [reply src/history/hs_rec.c:908]: Good point! Updated in [11b5985](https://github.com/wiredtiger/wiredtiger/pull/8268/commits/11b59858ef784730e3719fede5e8a840ecd1469d)

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Comment by ajmorton** [comment]
I've moved the assertion back to it original location. It turns out we do call this function with non-mixed-mode timestamps, but it won't find any newer updates in the history store and returns before reaching the assertion.
I think there's potential to optimise this code but I won't touch that in this PR.

**Review by quchenhao — APPROVED** [review]
lgtm

**Review by agorrod — APPROVED** [review]
lgtm

**Comment by ajmorton** [comment]
We have one failing test in the PR which is `test_stat08`. This is not caused by this PR and is already being fixed in WT-9774.

**Comment by ajmorton** [comment]
evergreen merge

**Pull request merged**